client/internal/peer/conn.go

```diff
@@ -399,6 +402,41 @@ func (conn *Conn) startProxy(remoteConn net.Conn, remoteWgPort int) error {
 	return nil
 }
 
+func (conn *Conn) getProxyWithMessageExchange(pair *ice.CandidatePair, remoteWgPort int) proxy.Proxy {
+	useProxy := shouldUseProxy(pair)
+
+	err := conn.sendSignalMessage(&sProto.Message{
+		Key:       conn.config.LocalKey,
+		RemoteKey: conn.config.Key,
+		Body: &sProto.Body{
+			Type: sProto.Body_MODE,
+			Mode: &sProto.Mode{
+				Direct: !useProxy,
+			},
+		},
+	})
+	if err != nil {
+		log.Errorf("got an error while sending the signal message with the connection mode, error: %s", err)
+	}
+
+	waitTimeout := time.After(time.Second)
+
+	select {
+	case receivedMSG := <-conn.modeMsg:
+		if receivedMSG.Direct != useProxy {
+			useProxy = true
+		}
+	case <-waitTimeout:
```

Comment:
This select cause an extra latency in the connection with older clients. Somehow the client does not publish his version number? If not I do not have better idea than this solution.

Reply:
newer versions are sending their own version in the signal message, I can introduce version validation for this method. See https://github.com/netbirdio/netbird/blob/main/signal/proto/signalexchange.proto#L54